### ISSUE TITLE
beamerpresenter embeds store paths in ~/.config and fails when they are gc-ed

### ISSUE THREAD

**Issue by symphorien** [NixOS/nixpkgs#194116]
### Describe the bug
After an update and a gc, beamerpresenter will fail with this message:
```
     0.058 CRITICAL unknown:0 - Could not read GUI config: "/nix/store/cnjrjbkg6mz24hk6bsmlgjmcjjg75sbv-beamerpresenter-0.2.0/etc/xdg/beamerpresenter/gui.json"
Slide label changes:   time old new
     0.104 CRITICAL unknown:0 - "Loading GUI config file failed for filename \"/nix/store/cnjrjbkg6mz24hk6bsmlgjmcjjg75sbv-beamerpresenter-0.2.0/etc/xdg/beamerpresenter/gui.json\". Using fallback GUI config file /nix/store/kfp440m9j1wvxcbxhypaxmp51wcq9900-beamerpresenter-0.2.2/etc/xdg/beamerpresenter/gui.json"
```

Current beamerpresenter is /nix/store/kfp440m9j1wvxcbxhypaxmp51wcq9900-beamerpresenter-0.2.2/bin/beamerpresenter but the error message is about the store path of an old beamerpresenter which was deleted by the gc.
This store path is actually stored in `~/.config/beamerpresenter/beamerpresenter.conf`
<details><summary>content</summary><pre>
[General]
cache%20pages=-1
frame%20time=50
gui%20config=/nix/store/cnjrjbkg6mz24hk6bsmlgjmcjjg75sbv-beamerpresenter-0.2.0/etc/xdg/beamerpresenter/gui.json
log=true
manual=/nix/store/cnjrjbkg6mz24hk6bsmlgjmcjjg75sbv-beamerpresenter-0.2.0/share/doc/beamerpresenter/README.html
memory=1.0486e+08

[drawing]
history%20length%20hidden=20
history%20length%20visible=50
mode=cumulative

[keys]
B=clear
Ctrl%2BO=open
Ctrl%2BQ=quit
Ctrl%2BS=save
Ctrl%2BShift%2BS=save as
Ctrl%2BShift%2BY=redo right
Ctrl%2BShift%2BZ=undo right
Ctrl%2BY=redo left, redo
Ctrl%2BZ=undo left, undo
End=last
F11=fullscreen
Home=first
P=toggle timer
PgDown=next
PgUp=previous
Shift%2BC=clear left, clear right
Shift%2BF5=next skipping overlays
Shift%2BPgUp=previous skipping overlays
Space=update

[rendering]
page%20part%20threshold=2.5
renderer=mupdf
rendering%20arguments=draw, -F%format, -w%width, -h%height, -o-, %file, %page
rendering%20command=/nix/store/cnjrjbkg6mz24hk6bsmlgjmcjjg75sbv-beamerpresenter-0.2.0/bin/mutool
</pre>
</details>

### Steps To Reproduce
1. run beamerpresenter
2. update nixpkgs
3. nix-store --delete the old beamer presenter
4. run the new one

### Expected behavior
works

### Screenshots

![image](https://user-images.githubusercontent.com/12595971/193472584-394792da-2a45-4ced-812b-b52da5583fd3.png)

### Notify maintainers

@pacien @dotlambda

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.70, NixOS, 22.05 (Quokka), 22.05.3300.82379884b2e`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"nixos-22.05, nixos-unstable"`
 - channels(symphorien): `"home-manager-22.05.tar.gz"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by pacien** [comment]
@stiglers-eponym: is there a way to avoid the default paths from being
written in ~/.config/beamerpresenter/beamerpresenter.conf?

It seems that the config keys "gui%20config", "manual", and
"rendering%20command" refer to Nix store paths corresponding to the
first version of BeamerPresenter which ran on a system. The
configuration file keeps pointing to the old package after an upgrade,
which is problematic when the old paths get garbage-collected.

**Comment by stiglers-eponym** [comment]
I've fixed this upstream, see #194203:
* these paths are now commented out in the config file by default
* invalid paths are removed from the config file when running BeamerPresenter
